### ISSUE TITLE
IniFile should not use ordinal Dictionary

### ISSUE THREAD

**Issue by AchimStuy** [fluffynuts/PeanutButter#46]
We have big issues with the [BOM](https://en.wikipedia.org/wiki/Byte_order_mark) at the beginning of the ini-Text-File being added to the first section (or first entry if there is no section header). So finding the sections or entrys fails, because `StringComparer.OrdinalIgnoreCase` also takes the BOM into account. Why are you using `OrdinalIgnoreCase`?
Could you either check for the BOM character or change the Dictionary to use `InvariantCultureIgnoreCase`?

https://github.com/fluffynuts/PeanutButter/blob/4cb5843a6ea83abb90a152e507e34b746add3dd8/source/INI/PeanutButter.INI/INIFile.cs#L438
[20210202.txt](https://github.com/fluffynuts/PeanutButter/files/5917755/20210202.txt)

**Comment by fluffynuts** [comment]
Hi, thanks for reporting an issue, and apologies for any inconvenience

Having just built a test, using your input file attached above, I can confirm that `StringComparer.OrdinalIgnoreCase` is _not_ the culprit. Although I agree that using `StringComparer.InvariantCultureIgnoreCase` is probably a better way to go, and have updated to use it, which breaks no existing tests. However the _new_ test I made for your case still fails, and it's because the string I'm seeing for the line that is being parsed has a leading character (the BOM) that is causing your first section to be stored with a name like so (debug view from Rider):

![image](https://user-images.githubusercontent.com/4669340/106738178-4b1b5080-6620-11eb-834d-1194b547b76b.png)

So I have to cater for this non-printable character and strip it off before processing the line (probably only for the first line to solve your particular issue). I'll find a fix and have an update shortly.

**Comment by fluffynuts** [comment]
A release should be out shortly to address this - please update to 2.0.5 when it's available and you have time & let me know.

**Comment by AchimStuy** [comment]
Thank you really much for your quick response! I'll let you know, when I have tested your release. :)

I was not sure, whether my issue was from the BOM-character or not. But I discovered the `Ordinal*`-Issue when I skipped the first section and just added a Key/Value-pair at the first line. There I expected to get the value via `GetValue(string.Empty, _key_)` but got an Exception. I don't remember the details, but I can investigate, if you want, though it should be fixed with your commit.
